androidtv/constants.py

```diff
@@ -28,20 +30,30 @@
 #: Determine whether the device is awake
 CMD_AWAKE = "dumpsys power | grep mWakefulness | grep -q Awake"
 
+#: Define current app
+CMD_DEFINE_CURRENT_APP = "CURRENT_APP=$(dumpsys window windows | grep -E 'mCurrentFocus|mFocusedApp') && " + VAR_CURRENT_APP_EXTRACTION
```

Comment:
This constant seems unnecessary. Same goes for the other `CMD_DEFINE_CURRENT_APP_*` constant.

Reply:
In which way do you see this as potentially unnecessary?

Besides the main objective to deduplicate the code, as the majority of the complexity is already defined in `VAR_CURRENT_APP_EXTRACTION`.

The other use case I had in mind, using this module from a Python script and being able to use `CURRENT_APP` (directly) in a custom command using **adb shell** without the additional (`CMD_CURRENT_APP`) command to obtain the `current_app`.

Reply:
My comments from earlier were based on looking at this pull request on my phone, and the "Find in page" functionality wasn't working right.  So I do see that these constants are used and that they make the code a bit more DRY.  

My main concerns are that

1. Adding more constants makes the pull request harder to review -- it's hard to see what is actually being changed.
2. Aside from this pull request, it's easier to read the code if the full command is spelled out:

```python
# I need to check what `VAR_CURRENT_APP_EXTRACTION` does
CMD_DEFINE_CURRENT_APP = "CURRENT_APP=$(dumpsys window windows | grep -E 'mCurrentFocus|mFocusedApp') && " + VAR_CURRENT_APP_EXTRACTION

# Here I know exactly what the ADB shell command is doing
CMD_CURRENT_APP = "CURRENT_APP=$(dumpsys window windows | grep mCurrentFocus) && CURRENT_APP=${CURRENT_APP#*{* * } && CURRENT_APP=${CURRENT_APP%%/*} && echo $CURRENT_APP"
```

> The other use case I had in mind, using this module from a Python script and being able to use `CURRENT_APP` (directly) in a custom command using **adb shell** without the additional (`CMD_CURRENT_APP`) command to obtain the `current_app`.

The way I see it, anyone doing that is a power user.  I think such a person would be more likely to look through these constants and piece together what they need in order to achieve their desired outcome OR simply formulate their ADB shell command from scratch rather than to pick out constants from this module.

Reply:
To assist with the review of the code, below is the diff of fully written commands by changed constant:

**CMD_CURRENT_APP**
```diff
-CURRENT_APP=$(dumpsys window windows | grep mCurrentFocus) && CURRENT_APP=${CURRENT_APP#*{* * } && CURRENT_APP=${CURRENT_APP%%/*} && echo $CURRENT_APP
+CURRENT_APP=$(dumpsys window windows | grep -E 'mCurrentFocus|mFocusedApp') && CURRENT_APP=${CURRENT_APP#*ActivityRecord{* * } CURRENT_APP=${CURRENT_APP#*{* * } && CURRENT_APP=${CURRENT_APP%%/*} && CURRENT_APP=${CURRENT_APP%\\}*} && echo $CURRENT_APP
```

**CMD_CURRENT_APP_GOOGLE_TV**
```diff
-CURRENT_APP=$(dumpsys activity a . | grep -E 'mResumedActivity' | cut -d ' ' -f 8) && CURRENT_APP=${CURRENT_APP%%/*} && echo $CURRENT_APP
+CURRENT_APP=$(dumpsys activity a . | grep mResumedActivity) && CURRENT_APP=${CURRENT_APP#*ActivityRecord{* * } CURRENT_APP=${CURRENT_APP#*{* * } && CURRENT_APP=${CURRENT_APP%%/*} && CURRENT_APP=${CURRENT_APP%\\}*} && echo $CURRENT_APP
```

**CMD_HDMI_INPUT**
```diff
-dumpsys activity starter | grep -o 'HDMIInputService\\/HW[0-9]' -m 1 | grep -o 'HW[0-9]'
+dumpsys activity starter | grep -E -o '(ExternalTv|HDMI)InputService/HW[0-9]' | grep -o 'HW[0-9]'
```

**CMD_LAUNCH_APP**
```diff
-CURRENT_APP=$(dumpsys window windows | grep mCurrentFocus) && CURRENT_APP=${{CURRENT_APP#*{{* * }} && CURRENT_APP=${{CURRENT_APP%%/*}} && if [ $CURRENT_APP != '{0}' ]; then monkey -p {0} -c " + INTENT_LAUNCH + " --pct-syskeys 0 1; fi
+CURRENT_APP=$(dumpsys window windows | grep -E 'mCurrentFocus|mFocusedApp') && CURRENT_APP=${{CURRENT_APP#*ActivityRecord{{* * }} CURRENT_APP=${{CURRENT_APP#*{{* * }} && CURRENT_APP=${{CURRENT_APP%%/*}} && CURRENT_APP=${{CURRENT_APP%\\}}*}} && if [ $CURRENT_APP != '{0}' ]; then monkey -p {0} -c " + INTENT_LAUNCH + ' --pct-syskeys 0 1; fi
```

**CMD_LAUNCH_APP_GOOGLE_TV**
```diff
-CURRENT_APP=$(dumpsys activity a . | grep -E 'mResumedActivity' | cut -d ' ' -f 8) && CURRENT_APP=${{CURRENT_APP%%/*}} && if [ $CURRENT_APP != '{0}' ]; then monkey -p {0} -c " + INTENT_LAUNCH + " --pct-syskeys 0 1; fi
+CURRENT_APP=$(dumpsys activity a . | grep mResumedActivity) && CURRENT_APP=${{CURRENT_APP#*ActivityRecord{{* * }} CURRENT_APP=${{CURRENT_APP#*{{* * }} && CURRENT_APP=${{CURRENT_APP%%/*}} && CURRENT_APP=${{CURRENT_APP%\\}}*}} && if [ $CURRENT_APP != '{0}' ]; then monkey -p {0} -c " + INTENT_LAUNCH + ' --pct-syskeys 0 1; fi
```

_Note_: If you plan to compare these commands via [android.adb_command service](https://www.home-assistant.io/integrations/androidtv/#androidtvadb_command) in Home Assistant, be sure to remove any escaped characters (e.g. `\\`, `{{`, `}}`) or to populate placeholders (`{0}`).

In regard to having more constants/variables, by simply looking at the above diffs you can see there is a lot of duplicate complexity, this should simplify any future changes to `CURRENT_APP` logic, as already mentioned in https://github.com/JeffLIrion/python-androidtv/pull/233#discussion_r571491318.